lib/ansible/modules/apt_repository.py

```diff
@@ -688,15 +708,17 @@ def main():
     sources_after = sourceslist.dump()
     changed = sources_before != sources_after
 
-    if changed and module._diff:
-        diff = []
-        for filename in set(sources_before.keys()).union(sources_after.keys()):
+    diff = []
+    sources_added = set()
+    sources_removed = set()
+    if changed:
+        sources_added = set(sources_after.keys()).difference(sources_before.keys())
+        sources_removed = set(sources_before.keys()).difference(sources_after.keys())
+        for filename in set(sources_added.union(sources_removed)):
```

Comment:
missing `if module._diff` we don't always want to return the file diffs unless asked for

Reply:
Check my new commit if i understood right